cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowContext.java

```diff
@@ -27,24 +26,29 @@
 
   WorkflowSpecification getWorkflowSpecification();
 
+  /**
+   * @throws UnsupportedOperationException if it is called from {@link Predicate}
```

Comment:
Since this is the part of the `WorkflowContext`, it is possible that the user might invoke this method through the predicate. Per discussion with @chtyim, throwing `UnsupportedOperationException` in that case.